python_modules/dagster/dagster/_core/definitions/parent_asset_graph_differ.py

```diff
@@ -0,0 +1,122 @@
+from enum import Enum
+from typing import TYPE_CHECKING, Callable, Optional, Sequence, Union
+
+import dagster._check as check
+from dagster._core.definitions.external_asset_graph import ExternalAssetGraph
+from dagster._core.workspace.context import BaseWorkspaceRequestContext
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.asset_graph import AssetGraph
+    from dagster._core.definitions.events import (
+        AssetKey,
+    )
+
+
+class ChangeReason(Enum):
+    NEW = "NEW"
+    CODE_VERSION = "CODE_VERSION"
+    INPUTS = "INPUTS"
+
+
+class ParentAssetGraphDiffer:
+    """Given two asset graphs, parent_asset_graph and branch_asset_graph, we can compute how the
+    assets in branch_asset_graph have changed with respect to parent_asset_graph. The ChangeReason
+    enum contains the list of potential changes an asset can undergo.
+    """
+
+    _branch_asset_graph: Optional["AssetGraph"]
+    _branch_asset_graph_load_fn: Optional[Callable[[], "AssetGraph"]]
+    _parent_asset_graph: Optional["AssetGraph"]
+    _parent_asset_graph_load_fn: Optional[Callable[[], "AssetGraph"]]
+
+    def __init__(
+        self,
+        branch_asset_graph: Union["AssetGraph", Callable[[], "AssetGraph"]],
+        parent_asset_graph: Optional[Union["AssetGraph", Callable[[], "AssetGraph"]]] = None,
+    ):
+        from dagster._core.definitions.asset_graph import AssetGraph
+
+        if parent_asset_graph is None:
+            # if parent_asset_graph is None, we are not in a branch deployment.
+            # TODO - need to confirm the behavior of a new code location in a branch deployment
```

Comment:
nit: personally find a bare `TODO` to be superfluous in comments like this. I find that they communicate just as effectively without that prefix without unclear expectations of who will do it when. I find that coming across an N years old TODO comment feels worse than finding the same comment without the TODO in front of it. I advocate for either dropping the todo or formalizing it more by like using a github issue link or other systemic tagging scheme.

Reply:
oh yeah this is more like a todo for me to figure out right now before this is mergable. Can remove because i put it as an item in the pr description

Reply:
resolved the todo and removed this